src/key.rs

```diff
@@ -228,8 +225,39 @@ impl PublicKey for RsaPublicKey {
 }
 
 impl RsaPublicKey {
-    /// Create a new key from its components.
+    /// Minimum value of the public exponent `e`.
+    pub const MIN_PUB_EXPONENT: u64 = 2;
+
+    /// Maximum value of the public exponent `e`.
+    pub const MAX_PUB_EXPONENT: u64 = (1 << 33) - 1;
```

Comment:
Feels a bit weird to make these `pub` as `u64`. I think it'd be nice to have access to them, but there is a mismatch between `u64` here and `BigUint` when passed as a parameter.

Perhaps we should make `e: u64` as well?